roles/etc_hosts/tasks/main.yml

```diff
@@ -1,10 +1,10 @@
 ---
 - name: "Build hosts file"
-  lineinfile:
-    dest: /etc/hosts
-    regexp: ".*{{ item.replace('.', '-') }}$"
-    line: "{{ hostvars[item].ansible_host }} {{ item.replace('.', '-') }}.{{ etc_hosts_domain }} {{ item.replace('.', '-') }}"
-    state: present
+  include_tasks: "./hosts_file_record.yml"
+  vars:
+    ip: "{{ hostvars[item].ansible_host }}"
+    domain: "{{ etc_hosts_domain }}"
+    hostname: "{{ item }}"
```

Comment:
if we pass `{{ item | regex_replace('\..*') }}` here instead (thus stripping the domain from the name), can we then avoid the duplicated code and complex searches/when conditions?

Reply:
wait, no, we can't as `item` will be something like `foreman-1.20.inecas.example.com`, right?

then `{{ item | regex_replace(etc_hosts_domain | regex_escape()) }}`?